stixcore/io/fits/processors.py

```diff
@@ -491,11 +490,21 @@ def write_fits(self, product):
             idb_hdu = table_to_hdu(idb_enc)
             idb_hdu.name = 'IDB_VERSIONS'
 
-            energy_enc = fits.connect._encode_mixins(energies)
-            energy_hdu = table_to_hdu(energy_enc)
-            energy_hdu.name = 'ENERGIES'
+            hdus = [primary_hdu, control_hdu, data_hdu, idb_hdu]
+
+            if getattr(prod, 'get_energies', False) is not False:
+                elow, ehigh = prod.get_energies()
+                energies = QTable()
+                energies['channel'] = range(len(elow))
+                energies['e_low'] = elow * u.keV
+                energies['e_high'] = ehigh * u.keV
+
+                energy_enc = fits.connect._encode_mixins(energies)
+                energy_hdu = table_to_hdu(energy_enc)
+                energy_hdu.name = 'ENERGIES'
+                hdus.append(energy_hdu)
 
-            hdul = fits.HDUList([primary_hdu, control_hdu, data_hdu, idb_hdu, energy_hdu])
+            hdul = fits.HDUList(hdus)
 
```

Comment:
was all this energy updates not already part of another PR

Reply:
Yea it was but it assumed the energy extension was always present I just made it optional.